lib/modules/apostrophe-schemas/index.js

```diff
@@ -260,12 +260,24 @@ module.exports = {
         // contain a method name. For bc don't mess with possible
         // existing usages in custom schema field types predating
         // this feature
-        field.moduleName = field.moduleName || (module && module.__meta.name);
+        self.setModuleName(field, module);
       });
 
       return schema;
     };
 
+    // Recursively set moduleName property of the field and any subfields,
+    // as might be found in array or object fields. `module` is an actual module
+
+    self.setModuleName = function(field, module) {
+      field.moduleName = field.moduleName || (module && module.__meta.name);
+      if ((field.type === 'array') || (field.type === 'object')) {
+        _.each(field.schema || [], function(subfield) {
```

Comment:
This is one of those places I'd expect Tom to tell me that it'd be better to have `field.schema || []` in parens for clarity and caution. This should be fine, though, so optional.